examples/trusted-sync/src/main.rs

```diff
@@ -39,6 +39,7 @@ async fn sync(cli: cli::Cli) -> Result<()> {
     let l1_rpc_url = cli.l1_rpc_url()?;
     let l2_rpc_url = cli.l2_rpc_url()?;
     let beacon_url = cli.beacon_url()?;
+    let blob_archiver_url = cli.blob_archiver_url()?;
```

Comment:
Even thought the argument is optional, this will cause the application to crash since the error is returned here.

We should introduce a builder pattern here IMO.
```
OnlineBlobProviderBuilder::new().with_beacon_client(beacon_client).with_fallback(Option<block_archiver_url>).build()?
```

This could return either the `OnlineBlobProvider` if no fallback is provided or `OnlineBlobProviderWithFallback` otherwise.

Reply:
Good point! Adding!